src/Actions/ImportProducts.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Rapidez\Statamic\Actions;
+
+use Statamic\Facades\Entry;
+use Rapidez\Statamic\Actions\CreateProducts;
+use Rapidez\Statamic\Actions\DeleteProducts;
+use Rapidez\Statamic\Events\ProductsImportedEvent;
+
+class ImportProducts
+{
+    public function __construct(
+        public CreateProducts $createProducts,
+        public DeleteProducts $deleteProducts
+    ) {
+    }
+
+    public function import(
+        ?string $store = null
+    ): void
+    {
+        $productModel = config('rapidez.models.product');
+        $storeModel = config('rapidez.models.store');
+        $storeModels = $store ? $storeModel::where('store_id', $store)->get() : $storeModel::all();
+
+        foreach($storeModels as $storeModel) {
+            config()->set('rapidez.store', $storeModel->store_id);
+
+            $products = $productModel::selectAttributes(['sku', 'name', 'url_key'])->where('catalog_product_flat_'.config('rapidez.store').'.visibility', 4)->get();
+            $storeModel = Entry::whereCollection('stores')->where('store_id', $storeModel->store_id)->first();
+
+            $products = $this->createProducts->create($products, $storeModel->id());
+            $this->deleteProducts->delete($products, $storeModel->id);
```

Comment:
We're importing and so creating products and directly after deleting them?

Reply:
Yes, we're passing the imported products to the delete function where we do a compare to check if the products in Statamic and the imported (Magento) products are the same. If there are any products in Statamic that do not exist in Magento, then we delete those products from Statamic. Maybe the naming was a bit confusing, changed it to deleteOldProducts().